docs/source/notes/extending_accelerator.md

```diff
@@ -0,0 +1,435 @@
+    :end-before: LITERALINCLUDE END: FALLBACK IMPL
+    :emphasize-lines: 2-5
+    :linenos:
+```
+
+::::
+
+Of course, global fallbacks can also be combined with a blacklist of fallbacks, which is a common approach, especially when only a few operators do not support fallbacks.
+
+#### PyTorch STUB
+
+PyTorch also provides another approach for built-in operators: `STUB`. This method is essentially based on the `Step 1<step-one>` approach, but adds secondary scheduling capabilities (for example, scheduling based on CPU characteristics).
+
+```{note}
+The `STUB` method currently supports only a limited set of operators. For new accelerator devices, the advantage of the `STUB` method is that it significantly reduces the cost of development at the cost of a small performance overhead. PyTorch currently does not clearly list the set of operators that can be registered through `STUB`. Due to the large number of related operators, only the query method for the supported operator list is provided here.
+```
+
+```shell
+pushd ${TORCH_ROOT}
+
+find aten -type f -a -name "*.h" | xargs -I {} grep -wl "^DECLARE_DISPATCH" {}
+
+popd
+```
+
+`DECLARE_DISPATCH` is a macro used to explicitly declare `STUB`. It is currently distributed in the `aten` directory. Based on this macro, you can find all operators that can be integrated using the `STUB` method.
+
+```text
+...
+aten/src/ATen/native/Activation.h
+aten/src/ATen/native/FusedSGD.h
+aten/src/ATen/native/nested/NestedTensorBinaryOps.h
+aten/src/ATen/native/TensorCompare.h
+aten/src/ATen/native/Sorting.h
+...
+```
+
+```c++
+using unary_fn = void(*)(TensorIteratorBase&);
+
+DECLARE_DISPATCH(unary_fn, abs_stub)
+```
+
+The above listing contains the file that declares the `STUB` operator, where you can clearly see the STUB name and the associated function signature. Next, we will take `abs_stub` as an example to briefly introduce the path to support operators through `STUB`.
+
+::::{tab-set}
+
+:::{tab-item} C++
+
+```{eval-rst}
+.. literalinclude:: ../../../test/cpp_extensions/open_registration_extension/torch_openreg/csrc/aten/native/Extra.cpp
+    :language: c++
+    :start-after: LITERALINCLUDE START: STUB ABS
+    :end-before: LITERALINCLUDE END: STUB ABS
+    :linenos:
+```
+
+```{eval-rst}
+.. literalinclude:: ../../../test/cpp_extensions/open_registration_extension/torch_openreg/csrc/aten/OpenRegExtra.cpp
+    :language: c++
+    :start-after: LITERALINCLUDE START: STUB DEFAULT
+    :end-before: LITERALINCLUDE END: STUB DEFAULT
+    :emphasize-lines: 1
+    :linenos:
+```
+
+:::
+
+::::
+
+From the signature, we can see that the input of `abs_stub` is `TensorIteratorBase`, a powerful helper class provided by PyTorch that contains all input and output operators, as well as some other auxiliary methods. Based on it, we can develop the `abs_kernel` operator and then call `REGISTER_PRIVATEUSE1_DISPATCH` to specify `abs_stub` to complete the registration.
+
+#### Custom Operators
```

Comment:
Please link to https://docs.pytorch.org/tutorials/advanced/custom_ops_landing_page.html for more advanced details on custom ops

Reply:
Done